### PR TITLE
DeployApplication - new procedure

### PR DESCRIPTION
**Pull request by ovasilko**

### CODE REVIEW

File: src/main/resources/project/server/deployApplication.pl [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+    my $params = $jboss->get_params_as_hashref(qw/
+        applicationContentSourcePath
+        deploymentName
+        runtimeName
+        enabledServerGroups
+        disabledServerGroups
+        additionalOptions
+        /);
+
+    my $param_application_content_source_path = $params->{applicationContentSourcePath};
```
Comment (justnoxx) [inline src/main/resources/project/server/deployApplication.pl]: Why do we need additional variables if we already have $params hashref?

File: src/main/resources/project/server/deployApplication.pl [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+        my @specified_server_groups_where_deployment_was_enabled = grep {
+            $all_specified_server_groups_hash{$_}
+                && $json_server_groups_discovery->{result}->{$_}->{deployment}
+                && $json_server_groups_discovery->{result}->{$_}->{deployment}->{$expected_deployment_name}
+                && $json_server_groups_discovery->{result}->{$_}->{deployment}->{$expected_deployment_name}->{enabled}
+        } @all_server_groups;
+        $jboss->log_info("FYI.. Deployment '$expected_deployment_name' is assigned and enabled on the server group(s) which are in enabled/disabled list: @specified_server_groups_where_deployment_was_enabled")
+            if @specified_server_groups_where_deployment_was_enabled;
+
+        my @specified_server_groups_where_deployment_was_disabled = grep {
```
Comment (justnoxx) [inline src/main/resources/project/server/deployApplication.pl]: Why do we need a bunch of these greps, if the logic is similar?

File: src/main/resources/project/server/deployApplication.pl [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+        );
+        $jboss->log_info("=======Finished: deploying to standalone=======");
+
+        my $summary = "Application '$expected_deployment_name' has been successfully deployed from '$expected_source_for_summary'";
+        $jboss->set_property(summary => $summary);
+        return;
+    }
+}
+
+sub run_command_and_get_json_with_exiting_on_error {
```
Comment (justnoxx) [inline src/main/resources/project/server/deployApplication.pl]: Maybe we need to move this code to JBoss.pm?

File: src/main/resources/project/server/deployApplication.pl (hunk 1 above)
Comment (ovasilko) [reply src/main/resources/project/server/deployApplication.pl]: It just for easier usage within IDE (exact auto completion by param perfix, without possible auto completion by keys). PS. Sure once we move to usage of many classes and their objects, we may forget about auto completion, but it is another story.

File: src/main/resources/project/server/deployApplication.pl (hunk 3 above)
Comment (ovasilko) [reply src/main/resources/project/server/deployApplication.pl]: It is possible way to go, only one thing that I do not want to make JBoss.pm bigger for now before understanding of it main purpose.

File: src/main/resources/project/server/deployApplication.pl (hunk 2 above)
Comment (ovasilko) [reply src/main/resources/project/server/deployApplication.pl]: there is different conditions in grep functions for each array. Some of them has almost the same, but "!" sign makes difference.

**Pull request merged**